### PR TITLE
feat(components): add JsonSchemaEditor with form/json/diff views

### PR DESCRIPTION
**Pull request by stevekinney**
### Summary
- Adds a full `JsonSchemaEditor` component with form, JSON, and diff workflows.
- Introduces robust state orchestration with guarded prop synchronisation (`readonly`, `draftOverride`) and lifecycle cleanup.
- Adds composable history via `useHistory` and integrates compile/meta validation with schema-draft-aware status updates.
- Replaces problematic composition row indexing and keyboard/ARIA interactions based on prior review fixes.
- Adds dedicated validator utilities, new unit tests, and playground examples.

### Committee roster
- typescript-expert
- testing-expert
- frontend-architect
- ux-designer
- simplicity-engineer
- junior-engineer
- security-reviewer
- codex

### Round
Round 2

### Tests
- bun test
- bun test packages/components/src/components/json-schema-editor/json-schema-editor-state.svelte.test.ts
- bun test packages/components/src/components/json-schema-editor/json-schema-validator.test.ts
- bun test packages/components/src/utilities/use-history.svelte.test.ts
- bun test packages/components/src/components/json-schema-editor/json-schema-editor.test.ts

---

> [!NOTE]
> **Medium Risk**
> Adds a sizable new editor surface plus new `ajv`-based validation and undo/redo state management; risk is mainly UI/state edge cases and validation performance/behavior across drafts.
> 
> **Overview**
> Adds a new `JsonSchemaEditor` component (exported via package `exports` and `src/index.ts`) that supports **form**, **raw JSON**, and **diff** workflows, including editor-level undo/redo/revert and copy-to-clipboard.
> 
> Introduces a new reactive `createEditorState` container with guarded prop synchronization (`readonly`, `draftOverride`, `schemaKey` reload), change/revert/validate callbacks, and debounced validation with a size gate to defer expensive AJV compilation for large drafts.
> 
> Implements form editing via recursive `PropertyEditor`/`PropertyList` (including stable keys for composition branches) and adds AJV-backed draft-aware meta-schema + compile checks (`json-schema-validator.ts`). Adds component styling, unit tests, and playground examples, and introduces `useHistory` as a new exported utility for generic undo/redo state.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 384a141c7f42ed923d9f2d986bdcb4370fd24746. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/components/src/components/json-schema-editor/json-schema-editor.svelte [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+  const state = createEditorState(stateOptions);
+
+  // Sync mutable props into the state container when the *prop* changes.
+  // We track the last applied value locally so the effect only calls the
+  // setter on real prop transitions — without that, the effect re-runs on
+  // every reactive read inside setReadonly/setDraftOverride and loops.
+  let lastReadonly = readonly;
+  let lastDraftOverride: JsonSchemaKnownDraft | undefined = draftOverride;
+  $effect(() => {
+    if (readonly !== lastReadonly) {
```
Comment (stevekinney) [inline packages/components/src/components/json-schema-editor/json-schema-editor.svelte:113]: Re-review: this addresses the earlier must-fix items around stale callback capture, live `readonly`/`draftOverride` resync, and unmount cleanup. I did not find a new issue in this area.
